clippy_lints/src/doc.rs

```diff
@@ -747,17 +747,22 @@ fn check_word(cx: &LateContext<'_>, word: &str, span: Span) {
         }
     }
 
-    // We assume that mixed-case words are not meant to be put inside bacticks. (Issue #2343)
+    // We assume that mixed-case words are not meant to be put inside backticks. (Issue #2343)
     if has_underscore(word) && has_hyphen(word) {
         return;
     }
 
     if has_underscore(word) || word.contains("::") || is_camel_case(word) {
-        span_lint(
+        let mut applicability = Applicability::MachineApplicable;
+
+        span_lint_and_sugg(
             cx,
             DOC_MARKDOWN,
             span,
-            &format!("you should put `{}` between ticks in the documentation", word),
+            "item in documentation is missing backticks",
+            "try",
+            format!("`{}`", snippet_with_applicability(cx, span, "..", &mut applicability)),
```

Comment:
There are needless backticks. This causes two backticks in suggestions.
```suggestion
            snippet_with_applicability(cx, span, "..", &mut applicability).into_owned(),
```

Reply:
These backticks are actually intentinal to make the suggestion machine applicable, but the suggestion sadly does look a bit strange :) See [comment](https://github.com/rust-lang/rust-clippy/pull/7904#discussion_r739807872)

Reply:
Oh, I got it, thanks!